doc/contribute.rst

```diff
@@ -0,0 +1,67 @@
+.. _contribute:
+
+
+Contribute documentation
+========================
+
+Documentation is an essential part of the software development process.
+We want to provide a useful piece of software. It is therefore necessary to
+have a good documentation, such that the user knows how to use our package.
+Contributions to the documentation are as welcome as contributions to the code.
+
+Docstrings
+----------
+
+We follow the numpy docstring standard.
+Check `here <https://github.com/numpy/numpy/blob/master/doc/HOWTO_DOCUMENT.rst.txt>`_ for a
+detailed explanation.
+
+
+Contribute tests
+================
+
+We're commited to testing our code. Tests that are required to pass are located in the
+``test`` folder. All files starting with ``test_`` contain tests and are automatically run
+on Travis CI. To run them manually, type::
+
+    python3 -m pytest test
+
+You can also run specific tests only.
+
+We encourage to test whatever possible. However, it might not always be easy to
+test code which is based on random sampling. We still encourage to provide general sanity
+and integration tests. We highly encourage a
+`test-driven development (TDD) <http://en.wikipedia.org/wiki/Test-driven_development>`_ style.
+
+Writing tests
+-------------
+
+Test can be written with `pytest <http://docs.pytest.org/en/latest/>`_
+or the `unittest <https://docs.python.org/3/library/unittest.html>`_ module.
+
+PEP8 Style Guide
+----------------
+
+We try to respect the `PEP8 <http://www.python.org/dev/peps/pep-0008/>`_ standard.
+We run `flake8 <http://flake8.pycqa.org/en/latest/>`_ as part of the test
+suite. The tests won't pass if flake8 complains.
+
+
+Contribute code
+===============
+
+If you start working on a new feature or a fix, f not already done, please
+create an issue on github, shortly describing your plans, and assign it to
+yourself. Your starting point should not be the master branch, but the
```

Comment:
I think you cannot block branching from master, but commiting/pushing/merging to master and also develop is now blocked, so that only accepted pull requests are allowed (unless you have magical admin powers to temporarily circumvent those things, of course).

Reply:
I exactly meant the commiting/pushing/merging :wink: